homeassistant/components/sensor/darksky.py

```diff
@@ -122,26 +131,39 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     sensors = []
     for variable in config[CONF_MONITORED_CONDITIONS]:
         sensors.append(DarkSkySensor(forecast_data, variable, name))
+        if forecast is not None and variable in ['temperature_min',
```

Comment:
After thinking about it some more I realized I misunderstood where most the data was coming from ('currently' instead of 'hourly'). I extended the SENSORS_TYPES sort of like you suggested. I added an array that lists which types the data comes from, 'currently', 'minutely', 'hourly', or 'daily'. I added daily forecast, but if someone wants to redo the config to make choosing which type of forecast they want for which fields, then the infrastructure is there.